lib/httpInterface.js

```diff
@@ -268,28 +269,29 @@ function handleGetListLookup(request, reply) {
   reply.raw.setHeader('Content-Type', replyType)
   const validate = getAjvResponseValidationFunction(request.routeSchema.response['200'].items)
 
-  this.crudService
-    .aggregate(crudContext, mongoQuery, projection, sort, skip, limit, stateArr, isTextSearchQuery)
-    .stream()
-    .on('error', (error) => {
-      log.error({ error }, 'Error during findAll stream')
-      // NOTE: error from Mongo may not serialize the message, so we force it here,
-      // we use debug level to prevent leaking potetially sensible information in logs.
-      log.debug({ error: { ...error, message: error.message } }, 'Error during findAll stream with message')
-
-      if (error.code === OPTIONS_INCOMPATIBILITY_ERROR_CODE) {
-        reply.getHttpError(BAD_REQUEST_ERROR_STATUS_CODE, error.message)
-        return
-      }
-      reply.getHttpError(INTERNAL_SERVER_ERROR_STATUS_CODE, error.message || 'something went wrong')
-    })
-    .pipe(this.castResultsAsStream())
-    .pipe(validateStream(validate))
-    .pipe(serializer())
-    .pipe(reply.raw)
+
+  try {
+    return await pipeline(
+      this.crudService
+        .aggregate(crudContext, mongoQuery, projection, sort, skip, limit, stateArr, isTextSearchQuery)
+        .stream(),
+      this.castResultsAsStream(),
+      validateStream(validate),
+      serializer(),
+      reply.raw
+    )
+  } catch (error) {
+    request.log.error({ error }, 'Error during findAll lookup stream')
+    // NOTE: error from Mongo may not serialize the message, so we force it here,
+    // we use debug level to prevent leaking potetially sensible information in logs.
+    request.log.debug({ error: { ...error, message: error.message } }, 'Error during findAll lookup stream with message')
```

Comment:
In case it is an error from Mongo, we may also log the `code` of the error

Reply:
It is already logged